lib/octopus_serverspec_extensions/type/octopus_deploy_project_group.rb

```diff
@@ -4,100 +4,123 @@
 
 module Serverspec::Type
   class OctopusDeployProjectGroup < Base
-    @projectgroup = nil
+    @project_group = nil
+    @project_group_name = nil
     @serverUrl = nil
     @apiKey = nil
     @serverSupportsSpaces = nil
     @spaceId = nil
     @spaceFragment = ""
 
-    def initialize(serverUrl, apiKey, projectgroup_name, space_name = nil)
-      @name = "Octopus Deploy Project Group #{projectgroup_name}"
+    def initialize(*url_and_api_key, project_group_name)
+      serverUrl, apiKey = get_octopus_creds(url_and_api_key)
+
+      @project_group_name = project_group_name
+
+      @name = "Octopus Deploy Project Group #{project_group_name}"
       @runner = Specinfra::Runner
       @serverUrl = serverUrl
       @apiKey = apiKey
 
-
-      if serverUrl.nil?
-        raise "'serverUrl' was not provided. Unable to connect to Octopus server to validate configuration."
-      end
-      if apiKey.nil?
-        raise "'apiKey' was not provided. Unable to connect to Octopus server to validate configuration."
-      end
-      if projectgroup_name.nil?
-        raise "'projectgroup_name' was not provided. Unable to connect to Octopus server to validate configuration."
-      end
-
       @serverSupportsSpaces = check_supports_spaces(serverUrl)
+    end
 
-      if @serverSupportsSpaces
-        # set the spaceId correctly
+    def exists?
+      load_resource_if_nil
+      (!@project_group.nil?) && (@project_group != [])
+    end
 
-        if space_name.nil?
-          @spaceId = 'Spaces-1' # default to Spaces-1
-        else
-          @spaceId = get_space_id?(space_name)
-        end
+    def has_description?(project_group_description)
+      load_resource_if_nil
+      return false if @project_group.nil?
+      @project_group["Description"] == project_group_description
+    end
 
-        @spaceFragment = "#{@spaceId}/"
+    def in_space(space_name)
+      # allows us to tag .in_space() onto the end of the resource. as in
+      # describe octopus_account("account name").in_space("MyNewSpace") do
+      @spaceId = get_space_id(space_name)
+      if @project_group_name.nil?
+        raise "'project_group_name' was not provided. Unable to connect to Octopus server to validate configuration."
```

Comment:
This seems odd... We've already connected to the server to get the space id?

Reply:
Nope. We only load the `@SpaceId` variable if we call `.in_space()` 

However there _is_ a duplicate `get_space_id()` function which I missed originally.